### ISSUE TITLE
Training accuracy

### ISSUE THREAD

**Issue by siddBanPsu** [pytorch/ignite#302]
In general, training accuracy is determined by the predictions during training and if there are dropout layers, those are in effect. So, generally we see training acc lower than val accuracy. However, in ignite, the evaluator has the metrics, when when runs after the epoch is complete, has dropout not effective and hence results in higher training acc. I saw there is an example that also says: "we do not want to encourage this pattern."on footnote of this page: https://pytorch.org/ignite/quickstart.html

I am wondering if this is the only way to get training metrics accurately without using the evaluator?

**Comment by vfdev-5** [comment]
@siddBanPsu precision of the model is lower if computed during the training because the model is also trained at the same time. The prediction on the first batch is more or less random (starting from scratch) and the prediction on the last batch is maybe better than previous ones. If total prediction is computed as an average on these predictions, we should understand that we do not measure the performance of the model with the last weights update.

Otherwise you can use [`RunningAverage`](https://github.com/pytorch/ignite/blob/master/ignite/metrics/running_average.py) that is smoothing metrics.

**Comment by siddBanPsu** [comment]
Thanks @vfdev-5 . I will use RunningAverage for the smoothing metrics.
